src/Microsoft.AspNetCore.Dispatcher/TemplateEndpoint.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Dispatcher
+{
+    public class TemplateEndpoint : Endpoint, ITemplateEndpoint
```

Comment:
There are a lot of constructors here. Don't think of this as something that's going to be in the face of most users. We will have a more polished startup experience.

Reply:
Maybe a builder pattern would be better.

Reply:
I built two different versions of that yesterday. I wanted to understand your ideas about application model before going further with it.